### PR TITLE
feat(models): [#FOR-362] add back models

### PR DESCRIPTION
**Pull request by qperie**
## Describe your changes
We added models for all Formulaire entity defined in BDD

## Issue ticket number and link
FOR-362 : https://jira.support-ent.fr/browse/FOR-362

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by Sofianebernoussi — APPROVED** [review]

**Review by Sofianebernoussi — APPROVED** [review]

File: common/src/main/java/fr/openent/form/core/models/Question.java [hunk 1]
```diff
@@ -1,136 +1,160 @@
...
+                .put(STATEMENT, this.statement)
+                .put(MANDATORY, this.mandatory)
+                .put(ORIGINAL_QUESTION_ID, this.originalQuestionId)
+                .put(SECTION_ID, this.sectionId)
+                .put(SECTION_POSITION, this.sectionPosition)
+                .put(CONDITIONAL, this.conditional)
+                .put(PLACEHOLDER, this.placeholder)
+                .put(MATRIX_ID, this.matrixId)
+                .put(MATRIX_POSITION, this.matrixPosition)
+                .put(QUESTION_CHOICES, new JsonArray(this.questionChoices))
```
Comment (qpic-cgi) [inline common/src/main/java/fr/openent/form/core/models/Question.java:154]: un peut dangereux ici:
this.questionChoices est de type QuestionChoice (et non json object), il faut  garantir que tu convertis bien l'élément en jsonObject:

```suggestion
// Dans Model.java :
default JsonArray toArray(List<I> models) {
        return new JsonArray(models.stream().map(Model::toJson).collect(Collectors.toList()));
    }
    
// ici, dans ta méthode toJson():
if (questionChoices != null) result.put(QUESTION_CHOICES, new QuestionChoice().toArray(questionChoices));
```

File: common/src/main/java/fr/openent/form/core/models/Question.java [hunk 2]
```diff
@@ -1,136 +1,160 @@
...
+                .put(CHILDREN, new JsonArray(this.children));
```
Comment (qpic-cgi) [inline common/src/main/java/fr/openent/form/core/models/Question.java:155]: pareil qu'au dessus ;)

File: common/src/main/java/fr/openent/form/core/models/Section.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+    public JsonObject toJson() {
+        return new JsonObject()
+                .put(ID, this.id)
+                .put(FORM_ID, this.formId)
+                .put(TITLE, this.title)
+                .put(POSITION, this.position)
+                .put(DESCRIPTION, this.description)
+                .put(ORIGINAL_SECTION_ID, this.originalSectionId)
+                .put(QUESTIONS, new JsonArray(this.questions));
```
Comment (qpic-cgi) [inline common/src/main/java/fr/openent/form/core/models/Section.java:65]: CF le commentaire sur la conversion de ta liste de models en liste de jsonObject

File: common/src/main/java/fr/openent/form/core/models/User.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
+package fr.openent.form.core.models;
+
+import io.vertx.core.json.JsonObject;
+
+import static fr.openent.form.core.constants.Fields.*;
+
+public class User implements Model<User> {
```
Comment (qpic-cgi) [inline common/src/main/java/fr/openent/form/core/models/User.java:7]: On va préférer l'étendre de UserInfo, pour éviter de réécrire ce qui existe déjà ;)

Ex:
https://github.com/OPEN-ENT-NG/minibadge/blob/dev/src/main/java/fr/openent/minibadge/model/User.java

**Review by qpic-cgi — CHANGES_REQUESTED** [review]

File: common/src/main/java/fr/openent/form/core/models/User.java (hunk 4 above)
Comment (qperie) [reply common/src/main/java/fr/openent/form/core/models/User.java:7]: C'est pas un object User de ce type là, c'est un User qui modélise les infos dans la table Users du système de partage. A la limite j'peux le renommer and ShareUser ou un truc comme ça si tu veux :)

**Review by qpic-cgi — APPROVED** [review]

**Pull request merged**